presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -630,6 +631,21 @@ private void addStatsListener(QueryExecution queryExecution)
     {
         Object lock = new Object();
 
+        AtomicBoolean queued = new AtomicBoolean();
+        queryExecution.addStateChangeListener(newValue -> {
```

Comment:
I put a breakpoint in the state change listener lambda and it doesn't get called  at all, can you double check please?